lib/runner.js

```diff
@@ -578,7 +578,7 @@ internals.protect = function (item, state) {
                 finish();
             }
             catch (ex) {
-                return finish(ex);
+                return finish(ex || 'reject');
```

Comment:
Shouldn't it be a `new Error()` ?

Reply:
If it is an `Error` then it will flow the full way through and report an empty string as the error, whereas setting it as a string results in the standard `Non Error object received or caught` message.

I am happy enough with either (my priority is that the test fail) so really depends on what feedback makes most sense

**with 'reject'**
```
Failed tests:

  27) Executor jobs throw:

      Non Error object received or caught

      at finish...
```

**with `new Error()`**
```
   27) Executor jobs throw:



      at Immediate.setImmediate...
```

Reply:
This seems reasonable to me, @Marsup any objections to the current implementation?

Reply:
OK, didn't investigate enough to tell, I was just surprised that it could end with an error or a string. If you're happy I'm happy.

Reply:
What's wrong with `new Error('Empty promise rejection.')`?

Reply:
As I said above, the "problem" with using an genuine `Error` is that you follow the same path as any errors that come from the test.

So you could see the error message `'Empty promise rejection.'` and think your test was actually throwing that.

The reason I chose a non-`Error` value is that I think the existing message `Non Error object received or caught` is the most appropriate and fits in nicely with the existing behaviour.

That said, as long as it is set to _something_ I do not care too much. My priority is that it fails the test.

Reply:
@geek if you want me to change anything just let me know